src/flir_adk_ethernet/EthernetCamera.cpp

```diff
@@ -123,6 +142,23 @@ bool EthernetCamera::camTypeMatches(string camType, INodeMap& nodeMapTLDevice) {
     return false;
 }
 
+bool EthernetCamera::setPTP() {
+    try {
+        INodeMap &nodeMap = _pCam->GetNodeMap();
+        CBooleanPtr ptpEnableNode = nodeMap.GetNode("GevIEEE1588");
+        CEnumerationPtr ptpModeNode = nodeMap.GetNode("GevIEEE1588Mode");
+
+        ptpEnableNode->SetValue(true);
```

Comment:
pretty sure nodes should be checked for validity before being set - check into that.

Reply:
they're currently not doing that in the package. may be a change for later / our driver?